crates/sui-types/src/crypto.rs

```diff
@@ -136,61 +136,56 @@ impl FromStr for SuiKeyPair {
 }
 
 impl EncodeDecodeBase64 for SuiKeyPair {
+    /// Encode a SuiKeyPair as `flag || pubkey || privkey` in Base64. The pubkey bytesare derived directly from privkey.
```

Comment:
**bytes _ are**

Reply:
Also, do we want for the base64 to have `flag || pubkey || privkey` or `flag || privkey` and then rederive the pub key as we do in the rest of serializers?

Reply:
I understand base64 is mostly for external UX, so I'm happy with both decisions, but we should document it properly as we received some feedback re different serializers or behavior in the past.

Reply:
IMO its more consistent to keep it `flag || pubkey || privkey` since the struct itself is called a keypair. As long as in our system, we read a keypair and ignores the pubkey part but to only read the privkey bytes to derive the keypair, i think its safe and less confusing. 

+1 lemme document this more explicitly in comments

Reply:
t depends on why we need such a serializer (cause encoding is usually used for (de)serialization). Assuming each receiver of the encoded format wants to deserialize the key pair into internal structs, then we should promote secure deserialization by rederiving the pub part for better hygiene, especially for this sensitive object type.

Re the argument "since the struct is called keypair" there are many types across many programming languages that their serialized bytes are compressed for a reason (security or bandwidth). I'd only buy the use case of compatibility with other ie JS libs to allow for quick conversion without rederivation, but still it's better to provide a tool to rederive Vs encoding it.

Reply:
It depends on why we need such a serializer (cause encoding is usually used for (de)serialization). Assuming each receiver of the encoded format wants to deserialize the key pair into internal structs, then we should promote secure deserialization by rederiving the pub part for better hygiene, especially sensitive object types.

Re the argument "since the struct is called keypair": there are many types across programming languages that their serialized bytes are compressed for a reason (security or bandwidth). I'd only buy the use case of compatibility with other (ie JS) libs to allow for quick conversion without rederivation, but still, IMO it's better to provide a tool to rederive Vs encoding it.

Reply:
revised.